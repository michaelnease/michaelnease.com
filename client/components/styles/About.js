import styled from "styled-components";

export const AboutStyled = styled.section`
  height: 80rem;
  position: relative;
  background: url("/static/images/bio/bio.gif") no-repeat center center;
  background-size: cover;
  border-bottom: 10px solid ${props => props.theme.colors.black};
  @media (max-width: 700px) {
    height: 100rem;
  }
`;

export const Text = styled.div`
  width: 60rem;
  p {
    text-align: justify;
    font-weight: 300;
    font-size: 1.7rem;
    margin-bottom: 20px;
  }
  @media (max-width: 700px) {
    padding: 3rem;
  }
`;

export const Rex = styled.img`
  box-shadow: ${props => props.theme.effects.bs};
  border: 1px solid ${props => props.theme.effects.black};
  float: left;
  width: 20rem;
  margin: ${props => props.theme.spacing.margin};
  position: relative;
  top: 10px;
`;
