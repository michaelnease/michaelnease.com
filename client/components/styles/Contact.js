import styled from "styled-components";

export const Form = styled.form`
  display: flex;
  position: relative;
  align-content: center;
  justify-content: center;
  background-color: ${props => props.theme.colors.darkGray};
  border-bottom: 10px solid ${props => props.theme.colors.black};
  padding: 3rem;
`;

export const Inputs = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  width: 80rem;
  grid-gap: 1.5rem;

  input,
  textarea {
    border: none;
    background-color: ${props => props.theme.colors.gray};
    padding-left: 3rem;
    font-size: 1.5rem;
    box-shadow: ${props => props.theme.effects.bs};
  }

  input {
    height: 5rem;
  }

  textarea {
    padding: 3rem;
    grid-column: span 2;
  }
`;
