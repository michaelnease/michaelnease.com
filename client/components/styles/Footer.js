import styled from "styled-components";

export const Foot = styled.footer`
  padding: 3rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 0 10rem;
  p {
    color: ${props => props.theme.colors.gray};
  }

  @media (max-width: 700px) {
    padding: 1rem;
    margin: 0;
  }
`;
