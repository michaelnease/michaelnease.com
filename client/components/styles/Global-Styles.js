import { createGlobalStyle } from "styled-components";

const GlobalStyles = createGlobalStyle`
  @font-face {
    font-family: 'Roboto';
    src: url('/static/fonts/Roboto-Regular.ttf') format('truetype');
    font-weight: normal;
    font-style: normal;
  }

  @font-face {
    font-family: 'Roboto';
    src: url('/static/fonts/Roboto-Light.ttf') format('truetype');
    font-weight: 300;
    font-style: normal;
  }

  html {
    box-sizing: border-box;
    font-size: 10px;
  }

  *, *:before, *:after {
    box-sizing: inherit;
  }

  body, h1, h2, h3, h4, h5, h6, p, ol, ul {
    margin: 0;
    padding: 0;
    font-weight: normal;
  }

  ol, ul {
    list-style: none;
  }

  img {
    max-width: 100%;
    height: auto;
  }

  body {
    padding: 0;
    margin: 0;
    font-size: 1.5rem;
    line-height: 2;
    font-family: "Roboto", Arial, Helvetica, sans-serif;
  }

  a {
    text-decoration: none;
  }
  
  button { font-family: "Roboto", Arial, Helvetica, sans-serif; }

  button:focus {
    outline:0;
  }

  textarea:focus, input:focus{
    outline: none;
  }
`;

export default GlobalStyles;
