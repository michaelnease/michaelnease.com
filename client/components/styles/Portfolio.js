import styled from "styled-components";
import Image from "../Image";

export const PortfolioStyled = styled.section`
  position: relative;
  border-bottom: 10px solid ${props => props.theme.colors.black};
`;

export const Work = styled.section`
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 20px;
  width: 80rem;
  margin: 10rem 0;
  @media (max-width: 700px) {
    padding: 0 1rem;
  }
`;

export const PortfolioImage = styled(Image)`
  grid-column: span ${props => props.span};
  border: 2px solid ${props => props.theme.colors.black};
  box-shadow: ${props => props.theme.effects.bs};
`;
