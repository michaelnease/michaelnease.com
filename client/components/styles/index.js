export * from "./Header";
export * from "./Quotes";
export * from "./About";
export * from "./Technology";
export * from "./Portfolio";
export * from "./Contact";
export * from "./Footer";
export * from "./Send-Message-Button";

export * from "./Texture";
