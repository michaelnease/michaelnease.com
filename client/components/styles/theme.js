export const theme = {
  colors: {
    black: "#141414",
    white: "#fff",
    gray: "#e0e0e0",
    darkGray: "#a9a9a9"
  },
  spacing: {
    margin: "15px",
    padding: "15px"
  },
  effects: {
    bs: "-5px 5px 5px 0px rgba(0, 0, 0, 0.2)"
  },
  bar: {
    height: "60px"
  }
};
