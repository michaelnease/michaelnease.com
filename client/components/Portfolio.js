import { PortfolioStyled, Work, Texture, PortfolioImage, Link } from './styles';
import { portfolio } from '../static/data/portfolio';

const Portfolio = () => {
  return (
    <PortfolioStyled>
      <Texture>
        <Link name="portfolio" />
        <Work>
          {portfolio.map(({ id, src, span }) => (
            <PortfolioImage key={id} src={src} span={span} />
          ))}
        </Work>
      </Texture>
    </PortfolioStyled>
  );
};

export default Portfolio;
