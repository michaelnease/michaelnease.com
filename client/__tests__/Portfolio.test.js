import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { render } from "react-testing-library";
import "jest-styled-components";
import { ThemeProvider } from "styled-components";

import { theme } from "../components/styles/theme";
import Portfolio from "../components/Portfolio";

configure({ adapter: new Adapter() });

describe("<Portfolio />", () => {
  const wrapper = shallow(<Portfolio />);

  it("Should render one <Work /> element", () => {
    expect(wrapper.find("Work").length).toBe(1);
  });

  it("Should render 7 <PortfolioImage /> elements", () => {
    expect(wrapper.find("PortfolioImage").length).toBe(9);
  });

  it("Should render <Work /> elements", () => {
    expect(
      wrapper
        .find("Work")
        .childAt(0)
        .prop("src")
    ).toBe("beyond-limits.gif");
    expect(
      wrapper
        .find("Work")
        .childAt(1)
        .prop("src")
    ).toBe("rentcom.gif");
    expect(
      wrapper
        .find("Work")
        .childAt(2)
        .prop("src")
    ).toBe("drumoff2011.jpg");
    expect(
      wrapper
        .find("Work")
        .childAt(3)
        .prop("src")
    ).toBe("drumoff2010.jpg");
    expect(
      wrapper
        .find("Work")
        .childAt(4)
        .prop("src")
    ).toBe("sessions.jpg");
    expect(
      wrapper
        .find("Work")
        .childAt(5)
        .prop("src")
    ).toBe("platinum-home.jpg");
    expect(
      wrapper
        .find("Work")
        .childAt(6)
        .prop("src")
    ).toBe("platinum-pdp.jpg");
    expect(
      wrapper
        .find("Work")
        .childAt(7)
        .prop("src")
    ).toBe("onlyincalifornia.jpg");
    expect(
      wrapper
        .find("Work")
        .childAt(8)
        .prop("src")
    ).toBe("powerlevelreview.jpg");
  });

  it("Should render <Work /> elements", () => {
    expect(
      wrapper
        .find("Work")
        .childAt(0)
        .prop("span")
    ).toBe(1);
    expect(
      wrapper
        .find("Work")
        .childAt(1)
        .prop("span")
    ).toBe(1);
    expect(
      wrapper
        .find("Work")
        .childAt(2)
        .prop("span")
    ).toBe(2);
    expect(
      wrapper
        .find("Work")
        .childAt(3)
        .prop("span")
    ).toBe(2);
    expect(
      wrapper
        .find("Work")
        .childAt(4)
        .prop("span")
    ).toBe(2);
    expect(
      wrapper
        .find("Work")
        .childAt(5)
        .prop("span")
    ).toBe(1);
  });

  it("Should match Snapshot", () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <Portfolio />
      </ThemeProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
