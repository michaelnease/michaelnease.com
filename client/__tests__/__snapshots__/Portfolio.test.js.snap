// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Portfolio /> Should match Snapshot 1`] = `
<DocumentFragment>
  .c2 {
  position: absolute;
  top: -60px;
}

.c0 {
  position: relative;
  border-bottom: 10px solid #141414;
}

.c3 {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 20px;
  width: 80rem;
  margin: 10rem 0;
}

.c4 {
  grid-column: span 2;
  border: 2px solid #141414;
  box-shadow: -5px 5px 5px 0px rgba(0,0,0,0.2);
}

.c5 {
  grid-column: span 1;
  border: 2px solid #141414;
  box-shadow: -5px 5px 5px 0px rgba(0,0,0,0.2);
}

.c1 {
  background-image: url("/static/images/common/texture.png");
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<section
    class="c0"
  >
    <div
      class="c1"
    >
      <a
        class="c2"
        name="portfolio"
      />
      <section
        class="c3"
      >
        <img
          class="c4"
          src="/static/images/portfolio/drumoff2011.jpg"
        />
        <img
          class="c4"
          src="/static/images/portfolio/drumoff2010.jpg"
        />
        <img
          class="c4"
          src="/static/images/portfolio/sessions.jpg"
        />
        <img
          class="c5"
          src="/static/images/portfolio/platinum-home.jpg"
        />
        <img
          class="c5"
          src="/static/images/portfolio/platinum-pdp.jpg"
        />
        <img
          class="c5"
          src="/static/images/portfolio/onlyincalifornia.jpg"
        />
        <img
          class="c5"
          src="/static/images/portfolio/powerlevelreview.jpg"
        />
      </section>
    </div>
  </section>
</DocumentFragment>
`;
